propolis/src/hw/pci/device.rs

```diff
@@ -1096,47 +1096,42 @@ impl MsixCfg {
                         }
                         MsixCapReg::TableOff => {
                             // table always at offset 0 for now
-                            ro.write_u32(0 | self.bar as u8 as u32);
+                            ro.write_u32(self.bar as u8 as u32);
                         }
                         MsixCapReg::PbaOff => {
                             ro.write_u32(self.pba_off | self.bar as u8 as u32);
                         }
                     }
                 }
                 RWOp::Write(wo) => {
-                    match id {
-                        MsixCapReg::MsgCtrl => {
```

Comment:
This was written as a match to be consistent with all of the other register handling